modules/tensor_mechanics/src/actions/CohesiveZoneMasterAction.C

```diff
@@ -45,19 +64,43 @@ CohesiveZoneMasterAction::act()
     for (unsigned int i = 0; i < displacements.size(); ++i)
     {
       // Create unique kernel name for each of the components
-      std::string unique_kernel_name = kernel_name + "_" + _name + "_" + Moose::stringify(i);
+      std::string unique_kernel_name = _czm_kernel_name + "_" + _name + "_" + Moose::stringify(i);
 
-      InputParameters paramsk = _factory.getValidParams(kernel_name);
+      InputParameters paramsk = _factory.getValidParams(_czm_kernel_name);
 
-      paramsk.set<bool>("use_displaced_mesh") = false;
       paramsk.set<unsigned int>("component") = i;
       paramsk.set<NonlinearVariableName>("variable") = displacements[i];
       paramsk.set<std::vector<VariableName>>("neighbor_var") = {displacements[i]};
       paramsk.set<std::vector<VariableName>>("displacements") = displacements;
       paramsk.set<std::vector<BoundaryName>>("boundary") =
           getParam<std::vector<BoundaryName>>("boundary");
 
-      _problem->addInterfaceKernel(kernel_name, unique_kernel_name, paramsk);
+      _problem->addInterfaceKernel(_czm_kernel_name, unique_kernel_name, paramsk);
     }
   }
+  else if (_current_task == "add_material")
+  {
+    // Create unique material name for the displacement jump provider
+    std::string unique_material_name = _disp_jump_provider_name + "_" + _name;
+    InputParameters paramsm = _factory.getValidParams(_disp_jump_provider_name);
+    paramsm.set<std::vector<BoundaryName>>("boundary") =
+        getParam<std::vector<BoundaryName>>("boundary");
+    paramsm.set<std::vector<VariableName>>("displacements") = displacements;
+    _problem->addInterfaceMaterial(_disp_jump_provider_name, unique_material_name, paramsm);
```

Comment:
Can we set different traction-separation law for different boundaries using the Action?

Reply:
need top try, I'll make a test for this

Reply:
The answer is yes.
I added a test for this case